packages/aws-cdk-lib/core/lib/nodejs-aspect.ts

```diff
@@ -0,0 +1,138 @@
+    }
+
+    //To handle providers
+    else if (construct instanceof Provider) {
+      this.handleProvider(construct);
+    }
+
+    //To handle single Function case
+    else if (construct instanceof ReceiptRuleSet) {
+      this.handleReceiptRuleSet(construct);
+    }
+  }
+
+  private handleCustomResourceProvider(provider: cdk.CustomResourceProviderBase): void {
+    const node = provider as cdk.CustomResourceProviderBase;
+    // only replace for nodejs case
+    if (this.isValidRuntime(node.runtime)) {
+      const targetnode = node.node.children.find((child) => child instanceof cdk.CfnResource &&
+      child.cfnResourceType === 'AWS::Lambda::Function') as cdk.CfnResource;
+      targetnode.addPropertyOverride('Runtime', 'nodejs20.x');
+    }
+  }
+
+  private handleProvider(provider: Provider): void {
+    const functionNodes = provider.node.children.filter((child) => child instanceof lambda.Function);
+    for ( var node of functionNodes) {
+      const targetNode = node.node.children.find((child) => child instanceof cdk.CfnResource && child.cfnResourceType === 'AWS::Lambda::Function') as cdk.CfnResource;
+      if (this.isValidRuntime(this.getRuntimeProperty(targetNode))) {
+        targetNode.addPropertyOverride('Runtime', 'nodejs20.x');
+      }
+
+      // onEvent Handlers
+      const onEventHandler = provider.onEventHandler as lambda.Function;
+      const onEventHandlerRuntime = onEventHandler.node.children.find((child) => child instanceof cdk.CfnResource && child.cfnResourceType === 'AWS::Lambda::Function') as cdk.CfnResource;
+      if (this.isValidRuntime(this.getRuntimeProperty(targetNode))) {
+        onEventHandlerRuntime.addPropertyOverride('Runtime', 'nodejs20.x');
+      }
+
+      //isComplete Handlers
+      // Handlers
+      const isCompleteHandler = provider.isCompleteHandler as lambda.Function;
+      const isCompleteHandlerRuntime = isCompleteHandler.node.children.find((child) => child instanceof cdk.CfnResource && child.cfnResourceType === 'AWS::Lambda::Function') as cdk.CfnResource;
+      if (this.isValidRuntime(this.getRuntimeProperty(targetNode))) {
+        isCompleteHandlerRuntime.addPropertyOverride('Runtime', 'nodejs20.x');
+      }
+    }
+  }
+
+  private handleReceiptRuleSet(ruleSet: ReceiptRuleSet): void {
+    const functionnode = ruleSet.node.findChild('DropSpam');
+    const ses = functionnode.node.findChild('Function') as lambda.CfnFunction;
+    if (ses.runtime && this.isValidRuntime(ses.runtime)) {
+      ses.addPropertyOverride('Runtime', 'nodejs20.x');
+    }
+  }
+
+  /**
+ *Runtime Validation
+ *
+ */
+  private isValidRuntime(runtime: string) : boolean {
+    return runtime === this.targetRuntime;
+  }
+
+  private getRuntimeProperty(node: cdk.CfnResource) {
+    return (node.getResourceProperty('runtime') || node.getResourceProperty('Runtime')) as string;
+  }
+}
+//}
+/**
+ * RuntimeAspect
+ */
+export class RuntimeAspect extends RuntimeAspectsBase {
```

Comment:
Let's opt for an enum-like class. This class is another layer of abstraction on top of `Runtime` which is an abstraction on top of `Aspect`. Let's enable users to use `Aspects.of(...).add(...)` with the aspect being created. I would see this looking like: `Aspects.of(scope).add(RuntimeAspect.nodeJs20())` or something similar.